lib/new_relic/agent/system_info.rb

```diff
@@ -19,6 +19,20 @@ def self.ruby_os_identifier
         RbConfig::CONFIG['target_os']
       end
 
+      def self.os_distribution
+        case
+        when darwin? then :darwin
+        when linux? then :linux
+        when bsd? then :bsd
+        when windows? then :windows
+        else ruby_os_identifier
+        end
+      end
+
+      def self.windows?
+        !!(ruby_os_identifier =~ /mingw/i || ruby_os_identifier =~ /mswin/i)
```

Comment:
You can have Ruby perform just a single regex match operation by combining the 2 strings with a pipe, and then directly change `=~` to `!~` to prevent having to negate (`!!`) the result:

```suggestion
        ruby_os_identifier !~ /(?:mingw|mswin)/i
```

Reply:
@fallwith I have checked your suggestion it doesn't give the expected output. Instead i have made a different optimisation [here](https://github.com/newrelic/newrelic-ruby-agent/pull/2779/commits/6a0fa469c14e5c3a80ff2181b1098ea4198d7cc9)

Reply:
Ah, sorry about that @prateeksen. I read the line of code too quickly. Given that you are in a method called `windows?`, you wanted `true` for Windows systems, not `false`. I should have suggested one of these:

- ruby_os_identifier =~ /(?:mingw|mswin)/i
- ruby_os_identifier.match?(/(?:mingw|mswin)/i)

Your approach of using `[]` works well too!